maint/gen_binding_c.py

```diff
@@ -5,28 +5,47 @@
 
 from local_python import MPI_API_Global as G
 from local_python.mpi_api import load_mpi_api
-from local_python.binding_c import dump_mpi_c, dump_Makefile_mk
+from local_python.binding_c import *
```

Comment:
it is not possible for me to do any meaningful review of this commit. there are 1500 lines of change. i don't know what to do with it besides look at the output.

Reply:
I know. Unless there is something you particularly notice, such as styles or obvious strangeness, I think we can merge them. They should not have impact to the current code other than the pt2pt chapter that already got merged. We'll review the output in the following chap-by-chap PRs and if there is errors, we can always add patches then.